frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -557,3 +569,226 @@ export default function EditorDialog(props: EditorDialogProps) {
 export function ViewDialog(props: Omit<EditorDialogProps, 'onSave'>) {
   return <EditorDialog {...props} onSave={null} />;
 }
+
+interface UploadYamlProps {
+  setUploadYaml: (value: boolean) => void;
+  setCode: React.Dispatch<React.SetStateAction<{ code: string; format: string }>>;
+}
+
+function UploadYourYaml(props: UploadYamlProps) {
```

Comment:
could you move this component to a different file? This one is quite large already